app/src/Controller/UserController.php

```diff
@@ -37,31 +38,93 @@ public function __construct(TranslatorInterface $translator, EntityManagerInterf
     }
 
     #[Route(path: '/local', name: 'users_local_index', methods: 'GET')]
-    public function indexUserLocal(UserRepository $userRepository): Response
-    {
+    public function indexUserLocal(
+        Request $request,
+        PaginatorInterface $paginator,
+        UserRepository $userRepository
+    ): Response {
+
+        $perPage = (int) $this->getParameter('app.per_page_global');
+        $perPage = $request->getSession()->has('perPage') ? $request->getSession()->get('perPage') : $perPage;
+
         /** @var User $user */
         $user = $this->getUser();
-        $adminUsers = $userRepository->searchByRole($user, '["ROLE_ADMIN"]');
-        $superAdminUsers = $userRepository->searchByRole($user, '["ROLE_SUPER_ADMIN"]');
-        $users = array_merge($adminUsers, $superAdminUsers);
+
+        $searchKey = $request->query->getString('search');
+
+        $users = $userRepository->search(
+            $user,
+            ['["ROLE_ADMIN"]', '["ROLE_SUPER_ADMIN"]'],
```

Comment:
Le controller ne devrait pas avoir à se soucier que les valeurs des rôles utilisateur sont stockées dans un tableau JSON. En plus, ça ne gère pas l’hypothétique cas d’un utilisateur qui possède 2 rôles (donc moins robuste).


```suggestion
            ['ROLE_ADMIN', 'ROLE_SUPER_ADMIN'],
```